src/azure-firewall/azext_firewall/custom.py

```diff
@@ -122,6 +153,31 @@ def update_azure_firewall(cmd, instance, tags=None, zones=None, private_ranges=N
         instance.additional_properties['Network.DNS.Servers'] = ','.join(dns_servers or '')
     if threat_intel_mode is not None:
         instance.threat_intel_mode = threat_intel_mode
+
+    if instance.hub_ip_addresses is None and hub_public_ip_addresses is not None:
+        raise CLIError('Cannot delete public ip addresses from vhub without creation.')
+    if hub_public_ip_count is not None:
+        try:
+            if instance.hub_ip_addresses.public_ips.count is not None and hub_public_ip_count > instance.hub_ip_addresses.public_ips.count:  # pylint: disable=line-too-long
+                instance.hub_ip_addresses.public_ips.count = hub_public_ip_count
+            else:
+                raise CLIError('Cannot decrease the count of hub ip addresses through --count.')
+        except AttributeError:
```

Comment:
When will this exception be raised?

Reply:
if user doesn't set --count before and this filed would be None.

Reply:
Do you mean instance.hub_ip_address is None?

---
In reply to: [446598636](https://github.com/Azure/azure-cli-extensions/pull/1937#discussion_r446598636) [](ancestors = 446598636)

Reply:
yes. Or `instance.hub_ip_address.public_ips` is None